Source/MLX/Documentation.docc/Organization/indexing.md

```diff
@@ -2,43 +2,75 @@
 
 Array subscripts.
 
-``MLXArray`` provides a number of subscript operators to provide access to the contents
-of an array.
+``MLXArray`` supports all the same indexing (see <doc:indexing>) as 
+the python `mx.array`, though in some cases they are written differently.
+In all cases both `MLXArray` and `mx.array` indexing strive to match
+[numpy indexing](https://numpy.org/doc/stable/user/basics.indexing.html).
```

Comment:
Big update to the indexing documentation:

- describe all the features
- explain how they are used in python
- and how that translates